src/namespace.rs

```diff
@@ -0,0 +1,37 @@
+use std::{borrow::Cow, ops::Deref};
+
+/// The namespace of a storage container. Meant to be constructed from "stringy" types.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct Ns(Cow<'static, [u8]>);
```

Comment:
If I'm not mistaken, this [`Cow`](https://doc.rust-lang.org/std/borrow/enum.Cow.html) thingy is similar (though more opaque / automatic) and could be used to replace our [`Key`](https://github.com/CosmWasm/cw-storage-plus/blob/96f26bf06878d8a3d6f780cdc94ee10ef6debfc7/src/keys.rs#L6-L14) enum.

Moreover: Perhaps this `Ns` and `Key` can be conflated into a single thing. In another PR?